app/services/create_new_device_alert.rb

```diff
@@ -0,0 +1,33 @@
+class CreateNewDeviceAlert < ApplicationJob
+  include UserAlerts
+  queue_as :long_running
+
+  def perform(now)
+    emails_sent = 0
+    User.where(
+      sql_query_for_users_with_new_device,
+      tvalue: now - IdentityConfig.store.new_device_alert_delay_in_minutes.minutes,
+    ).each do |user|
+      emails_sent += 1 if clear_new_device_and_send_email(user)
+    end
+
+    emails_sent
+  end
+
+  private
+
+  def sql_query_for_users_with_new_device
+    <<~SQL
+      sign_in_new_device_at < :tvalue AND
+      sign_in_new_device_at IS NOT NULL
+    SQL
+  end
+
+  def clear_new_device_and_send_email(user)
+    user.sign_in_new_device_at = nil
+    user.save
```

Comment:
* This should probably happen as part of `AlertUserAboutNewDevice#send_alert`, since we already have logic for assigning this value as part of the initial `call` there
* We could collapse this to a single line using `user.update(sign_in_new_device_at: nil)`
* What's the reasoning for using `save` here but `save!` in `AlertUserAboutNewDevice.call` ?

Reply:
That makes sense! 
I may have misapplied the 'bang' there. I'm guessing it's unnecessary here. What do you think?
https://stackoverflow.com/questions/5056385/when-to-use-save-vs-save-in-model

Reply:
I'd think we can probably let it not throw, and maybe incorporate successful update as part of the incrementing of `emails_sent` (i.e. when we wire this up, make `send_alert` return a boolean depending if it successfully sent and updated the user).